crates/task-impls/src/quorum_proposal/dependency_handle.rs

```diff
@@ -185,6 +187,13 @@ impl<TYPES: NodeType> HandleDepOutput for ProposalDependencyHandle<TYPES> {
             .validated_state_map()
             .contains_key(&high_qc_view_number)
         {
+            // The proposal for the high qc view is missing, try to get it asynchronously
+            let memberhsip = Arc::clone(&self.quorum_membership);
+            let sender = self.sender.clone();
+            let consensus = Arc::clone(&self.consensus);
+            async_spawn(async move {
+                fetch_proposal(high_qc_view_number, sender, memberhsip, consensus).await
```

Comment:
Looks like this acquires a write lock on each invocation. Are we worried about accidentally holding a lock while something else needs it?